pitop/processing/algorithms/ball_detect.py

```diff
@@ -36,40 +36,60 @@ def __init__(self, color):
         }
         self.match_limit = match_limits[color]
 
-        self.center_points = deque(maxlen=DETECTION_POINTS_BUFFER_LENGTH)
-        self.radius = 0
-        self.angle_from_center = None
+        self._center_points_cv = deque(maxlen=DETECTION_POINTS_BUFFER_LENGTH)
+        self._center = None
+        self._radius = 0
+        self._angle = None
 
-    def clear(self):
-        self.center_points.clear()
+    def no_detection(self):
+        self.center_points_cv.appendleft(None)
+        self.center = None
         self.radius = 0
-        self.angle_from_center = None
+        self.angle = None
+
+    @property
+    def angle(self) -> float:
+        return self._angle
+
+    @angle.setter
+    def angle(self, value: float):
+        self._angle = value
 
     @property
-    def angle(self):
-        return self.angle_from_center
+    def center(self) -> tuple:
+        return self._center
+
+    @center.setter
+    def center(self, value: tuple):
+        self._center = value
+
+    @property
+    def radius(self) -> int:
+        return self._radius
+
+    @radius.setter
+    def radius(self, value: int):
```

Comment:
Why use getters and setters like this if they behave the same way as a member variable?

Reply:
As I was interacting with it in the test file I thought it would be better to have a more defined interface, would also provide return value type checks for people interacting with the class in an IDE

Reply:
Okay - let's add docstrings then because - as I have highlighted in another comment - it's not entirely clear what they all represent.